### ISSUE TITLE
Naming a slot "items" causes infinite loop

### ISSUE THREAD

**Issue by rhysb27** [ViewComponent/view_component#1604]
### Steps to reproduce
1. Create a component with a passthrough slot named "items":
  ```ruby
  renders_many :items, "ItemComponent"

  # ...

  class ItemComponent < ViewComponent::Base
    def call
      content
    end
  end
```

2. In a view, instantiate the component and fill the slot one or more times:
```erb
<% foo.bar(**args) do |bar| %>
  <%= bar.item do %>
    <p>Hello World!</p>
  <% end %>
<% end %>
```

3. Add some logging statements to the view and component to see the behaviour despite unsuccesful render.

### Expected behavior
The component should render the slot the specified number of times.

### Actual behavior
Upon render, the component enters into an infinite loop, eventually crashing the Rails server.

### System configuration
**Rails version**: 7.0.4

**Ruby version**: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]

**Gem version**: 2.76.0

**Comment by Spone** [comment]
Hi! Thanks for taking the time to open an issue. I'm wondering if something else could cause the infinite loop in your code, because I've been using slots named `items` on multiple occasions and I've never encountered this bug.

Do you think you can provide an example repo as explained [here](https://github.com/ViewComponent/view_component/blob/main/docs/CONTRIBUTING.md#reporting-bugs)?

**Comment by rhysb27** [comment]
Hi @Spone , thanks for replying so quickly!

I've been attempting to replicate the issue in an example repo but appear unable to. It's strange, when I encountered the bug I narrowed it down to being an issue with components, and all I did to fix it was change the name of the slot. 

Maybe the bug is more specific than I thought and is actually a consequence of two gems not playing nicely together. I could take the example repo further by adding in SimpleForm, but at that point it's not really an issue with ViewComponent anymore. If you're interested, the full pattern I was using was this:
```erb
<%= simple_form_for foo do |form| %>
  ...
  <%= render NestedFieldsetGroupComponent.new do |group| %>
    <% form.simple_fields_for :bar do |nested_form| %>
      <% group.item do %>
        ...
      <% end %>
    <% end %>
  <% end %>
<% end %>
```

Since I've been unable to replicate this issue so far I'm happy for the issue to be closed or to provide more information if it's something you'd like to look into further.
